lib/plugins/aws/package/compile/events/httpApi/index.js

```diff
@@ -48,9 +52,34 @@ class HttpApiEvents {
       },
     };
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8021
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'httpApi', {
-      anyOf: [{ type: 'string' }, { type: 'object' }],
+      oneOf: [
+        { type: 'string', regexp: methodPathPattern.toString() },
+        {
+          type: 'object',
+          properties: {
+            method: { type: 'string', regexp: methodPattern.toString() },
+            path: { type: 'string', minLength: 1 },
```

Comment:
For consistency we should also add here `pattern: "^(?:\\*|\\/\\S*)$"` (and then `minLength` can be removed)